### PR TITLE
update changelog and version number for v2.7.0

### PR DESCRIPTION
**Pull request by avpeery**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,26 @@
+## 2.7.0-beta.1
+
+## Features ✨ and improvements 🏁
```
Comment (karimnaaji) [inline CHANGELOG.md:3]:
```suggestion
### Features ✨ and improvements 🏁
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,26 @@
...
+
+* Enable preloading tiles for camera animation. ([#11328](https://github.com/mapbox/mapbox-gl-js/pull/11328))
+* Add perspective correction for non-rectangular image, canvas and video sources. ([#11292](https://github.com/mapbox/mapbox-gl-js/pull/11292))
+* Add function call chaining to function `map.setProjection`. ([#11279](https://github.com/mapbox/mapbox-gl-js/pull/11279)) (h/t [lpizzinidev](https://github.com/lpizzinidev))
+* Improve quality of transparent line layers by removing overlapping geometry artifacts. ([#11082](https://github.com/mapbox/mapbox-gl-js/pull/11082))
+* Improve performance of default markers. ([#11321](https://github.com/mapbox/mapbox-gl-js/pull/11321))
+* Add marker methods `setSnapToPixel` and `getSnapToPixel` to indicate rounding a marker to pixel. (h/t [@malekeym](https://github.com/malekeym)) ([#11167](https://github.com/mapbox/mapbox-gl-js/pull/11167)) (h/t [malekeym](https://github.com/malekeym))
+* Add a default aria-label for interactive markers for improved user accessibility. ([#11349](https://github.com/mapbox/mapbox-gl-js/pull/11349))
```
Comment (karimnaaji) [inline CHANGELOG.md:11]:
```suggestion
Add a default `aria-label` for interactive markers for improved user accessibility. ([#11349](https://github.com/mapbox/mapbox-gl-js/pull/11349))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,26 @@
+## 2.7.0-beta.1
+
+## Features ✨ and improvements 🏁
+
+* Enable preloading tiles for camera animation. ([#11328](https://github.com/mapbox/mapbox-gl-js/pull/11328))
+* Add perspective correction for non-rectangular image, canvas and video sources. ([#11292](https://github.com/mapbox/mapbox-gl-js/pull/11292))
```
Comment (karimnaaji) [inline CHANGELOG.md:6]: We try to give the changelog some order of importance based on the change. I think this one could be reordered further down as it's relatively minor compared to changes referenced further below. A possible order suggestion (feel free to adjust if you think there's a better one):

- preloading of camera animations
- improved transparency of lines
- improved performance of markers
- improved accessibility
- custom attribution order
- function chaining update to `map.setProjection`

I'd apply similar rearranging to the bug fix list as well.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,26 @@
...
+* Allow users to set order of custom attribution. ([#11196](https://github.com/mapbox/mapbox-gl-js/pull/11196))
+
+### 🐞 Bug fixes
+
+* Fix canvas size to evaluate to expected value when applying the CSS transform property. ([#11310](https://github.com/mapbox/mapbox-gl-js/pull/11310))
+* Fix rendering of denormalized strings with diacritics. ([#11269](https://github.com/mapbox/mapbox-gl-js/pull/11269))
+* Fix a `getBounds` issue that sometimes causes returning invalid LngLat when zooming on a map with terrain. ([#11339](https://github.com/mapbox/mapbox-gl-js/pull/11339))
+* Fix a rare terrain flickering issue when using terrain with multiple vector data sources. ([#11346](https://github.com/mapbox/mapbox-gl-js/pull/11346))
+* Fix returning 404s with sparse terrain DEM tilesets. ([#11276](https://github.com/mapbox/mapbox-gl-js/pull/11276))
+* Fix Safari 15 warning: `TypeError: undefined is not an object (evaluating 't[12]')`. ([#11330](https://github.com/mapbox/mapbox-gl-js/pull/11330))
```
Comment (karimnaaji) [inline CHANGELOG.md:21]: Is there more context we can give to this one. In what circumstances did this get triggered? Is it a relatively rare bug? If so, maybe we should note it.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,26 @@
...
+* Add a default aria-label for interactive markers for improved user accessibility. ([#11349](https://github.com/mapbox/mapbox-gl-js/pull/11349))
+* Allow users to set order of custom attribution. ([#11196](https://github.com/mapbox/mapbox-gl-js/pull/11196))
+
+### 🐞 Bug fixes
+
+* Fix canvas size to evaluate to expected value when applying the CSS transform property. ([#11310](https://github.com/mapbox/mapbox-gl-js/pull/11310))
+* Fix rendering of denormalized strings with diacritics. ([#11269](https://github.com/mapbox/mapbox-gl-js/pull/11269))
+* Fix a `getBounds` issue that sometimes causes returning invalid LngLat when zooming on a map with terrain. ([#11339](https://github.com/mapbox/mapbox-gl-js/pull/11339))
+* Fix a rare terrain flickering issue when using terrain with multiple vector data sources. ([#11346](https://github.com/mapbox/mapbox-gl-js/pull/11346))
+* Fix returning 404s with sparse terrain DEM tilesets. ([#11276](https://github.com/mapbox/mapbox-gl-js/pull/11276))
```
Comment (karimnaaji) [inline CHANGELOG.md:20]: This one almost seems like a feature, how about moving it there and reword to? cc @stepankuzmin 
```suggestion
* Add support for sparse tile sets to DEM data sources, when tiles don't go up to the full `maxzoom`. ([#11276](https://github.com/mapbox/mapbox-gl-js/pull/11276))
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,26 @@
...
+* Improve performance of default markers. ([#11321](https://github.com/mapbox/mapbox-gl-js/pull/11321))
+* Add marker methods `setSnapToPixel` and `getSnapToPixel` to indicate rounding a marker to pixel. (h/t [@malekeym](https://github.com/malekeym)) ([#11167](https://github.com/mapbox/mapbox-gl-js/pull/11167)) (h/t [malekeym](https://github.com/malekeym))
+* Add a default aria-label for interactive markers for improved user accessibility. ([#11349](https://github.com/mapbox/mapbox-gl-js/pull/11349))
+* Allow users to set order of custom attribution. ([#11196](https://github.com/mapbox/mapbox-gl-js/pull/11196))
+
+### 🐞 Bug fixes
+
+* Fix canvas size to evaluate to expected value when applying the CSS transform property. ([#11310](https://github.com/mapbox/mapbox-gl-js/pull/11310))
+* Fix rendering of denormalized strings with diacritics. ([#11269](https://github.com/mapbox/mapbox-gl-js/pull/11269))
+* Fix a `getBounds` issue that sometimes causes returning invalid LngLat when zooming on a map with terrain. ([#11339](https://github.com/mapbox/mapbox-gl-js/pull/11339))
```
Comment (karimnaaji) [inline CHANGELOG.md:18]: Suggestion for brevity:
```suggestion
* Fix `getBounds` sometimes returning invalid `LngLat` when zooming on a map with terrain. ([#11339](https://github.com/mapbox/mapbox-gl-js/pull/11339))
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,26 @@
...
+
+### Features ✨ and improvements 🏁
+
+* Enable preloading tiles for camera animation. ([#11328](https://github.com/mapbox/mapbox-gl-js/pull/11328))
+* Add perspective correction for non-rectangular image, canvas and video sources. ([#11292](https://github.com/mapbox/mapbox-gl-js/pull/11292))
+* Add function call chaining to function `map.setProjection`. ([#11279](https://github.com/mapbox/mapbox-gl-js/pull/11279)) (h/t [lpizzinidev](https://github.com/lpizzinidev))
+* Improve quality of transparent line layers by removing overlapping geometry artifacts. ([#11082](https://github.com/mapbox/mapbox-gl-js/pull/11082))
+* Improve performance of default markers. ([#11321](https://github.com/mapbox/mapbox-gl-js/pull/11321))
+* Add marker methods `setSnapToPixel` and `getSnapToPixel` to indicate rounding a marker to pixel. (h/t [@malekeym](https://github.com/malekeym)) ([#11167](https://github.com/mapbox/mapbox-gl-js/pull/11167)) (h/t [malekeym](https://github.com/malekeym))
+Add a default `aria-label` for interactive markers for improved user accessibility. ([#11349](https://github.com/mapbox/mapbox-gl-js/pull/11349))
```
Comment (karimnaaji) [inline CHANGELOG.md:11]: Missing bullet:
```suggestion
* Add a default `aria-label` for interactive markers for improved user accessibility. ([#11349](https://github.com/mapbox/mapbox-gl-js/pull/11349))
```

File: CHANGELOG.md (hunk 4 above)
Comment (avpeery) [reply CHANGELOG.md:21]: It was a rare webkit bug solved by a different for loop - maybe unnecessary to include in changelog

File: CHANGELOG.md (hunk 5 above)
Comment (avpeery) [reply CHANGELOG.md:20]: Seems good to me! Thanks for helping with rewording!

File: CHANGELOG.md (hunk 4 above)
Comment (karimnaaji) [reply CHANGELOG.md:21]: Yes you're right, sounds good skipping from what I recall of this issue! Seems very rare, platform and version specific.

File: CHANGELOG.md (hunk 3 above)
Comment (karimnaaji) [reply CHANGELOG.md]: Sorry, I meant to reference the `Add function call chaining to function `map.setProjection`. ([#11279](https://github.com/mapbox/mapbox-gl-js/pull/11279)) (h/t [lpizzinidev](https://github.com/lpizzinidev))` entry. Not perspective correction one!

Comment (avpeery) [reply CHANGELOG.md]: got it!

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,26 @@
...
+* Add perspective correction for non-rectangular image, canvas and video sources. ([#11292](https://github.com/mapbox/mapbox-gl-js/pull/11292))
+ox-gl-js/pull/11279)) (h/t [lpizzinidev](https://github.com/lpizzinidev))
+* Improve performance of default markers. ([#11321](https://github.com/mapbox/mapbox-gl-js/pull/11321))
+* Add marker methods `setSnapToPixel` and `getSnapToPixel` to indicate rounding a marker to pixel. (h/t [@malekeym](https://github.com/malekeym)) ([#11167](https://github.com/mapbox/mapbox-gl-js/pull/11167)) (h/t [malekeym](https://github.com/malekeym))
+* Add a default `aria-label` for interactive markers for improved user accessibility. ([#11349](https://github.com/mapbox/mapbox-gl-js/pull/11349))
+* Add support for sparse tile sets to DEM data sources, when served tiles don't go up to the full `maxzoom`. ([#11276](https://github.com/mapbox/mapbox-gl-js/pull/11276))
+* Allow users to set order of custom attribution. ([#11196](https://github.com/mapbox/mapbox-gl-js/pull/11196))
+* Add function call chaining to function `map.setProjection`. ([#11279](https://github.com/mapbox/mapb
```
Comment (karimnaaji) [inline CHANGELOG.md:14]: This line got stripped out:
```suggestion
* Add function call chaining to function `map.setProjection` ([#11279](https://github.com/mapbox/mapbox-gl-js/pull/11279)) (h/t [lpizzinidev](https://github.com/lpizzinidev))
```

**Review by karimnaaji — APPROVED** [review]
Looks good to me.

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,26 @@
+## 2.7.0-beta.1
+
+### Features ✨ and improvements 🏁
+
+* Enable preloading tiles for camera animation. ([#11328](https://github.com/mapbox/mapbox-gl-js/pull/11328))
+* Improve quality of transparent line layers by removing overlapping geometry artifacts. ([#11082](https://github.com/mapbox/mapbox-gl-js/pull/11082))
+* Add perspective correction for non-rectangular image, canvas and video sources. ([#11292](https://github.com/mapbox/mapbox-gl-js/pull/11292))
+ox-gl-js/pull/11279)) (h/t [lpizzinidev](https://github.com/lpizzinidev))
+* Improve performance of default markers. ([#11321](https://github.com/mapbox/mapbox-gl-js/pull/11321))
+* Add marker methods `setSnapToPixel` and `getSnapToPixel` to indicate rounding a marker to pixel. (h/t [@malekeym](https://github.com/malekeym)) ([#11167](https://github.com/mapbox/mapbox-gl-js/pull/11167)) (h/t [malekeym](https://github.com/malekeym))
```
Comment (SnailBones) [inline CHANGELOG.md:9]: Double h/t
```suggestion
* Add marker methods `setSnapToPixel` and `getSnapToPixel` to indicate rounding a marker to pixel. ([#11167](https://github.com/mapbox/mapbox-gl-js/pull/11167)) (h/t [malekeym](https://github.com/malekeym))
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,26 @@
+## 2.7.0-beta.1
+
+### Features ✨ and improvements 🏁
+
+* Enable preloading tiles for camera animation. ([#11328](https://github.com/mapbox/mapbox-gl-js/pull/11328))
+* Improve quality of transparent line layers by removing overlapping geometry artifacts. ([#11082](https://github.com/mapbox/mapbox-gl-js/pull/11082))
+* Add perspective correction for non-rectangular image, canvas and video sources. ([#11292](https://github.com/mapbox/mapbox-gl-js/pull/11292))
+ox-gl-js/pull/11279)) (h/t [lpizzinidev](https://github.com/lpizzinidev))
```
Comment (SnailBones) [inline CHANGELOG.md]: This test seems lost, I think it belongs down below

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,26 @@
...
+* Add marker methods `setSnapToPixel` and `getSnapToPixel` to indicate rounding a marker to pixel. (h/t [@malekeym](https://github.com/malekeym)) ([#11167](https://github.com/mapbox/mapbox-gl-js/pull/11167)) (h/t [malekeym](https://github.com/malekeym))
+* Add a default `aria-label` for interactive markers for improved user accessibility. ([#11349](https://github.com/mapbox/mapbox-gl-js/pull/11349))
+* Add support for sparse tile sets to DEM data sources, when served tiles don't go up to the full `maxzoom`. ([#11276](https://github.com/mapbox/mapbox-gl-js/pull/11276))
+* Allow users to set order of custom attribution. ([#11196](https://github.com/mapbox/mapbox-gl-js/pull/11196))
+* Add function call chaining to function `map.setProjection`. ([#11279](https://github.com/mapbox/mapb
+
+### 🐞 Bug fixes
+
+* Fix canvas size to evaluate to expected value when applying the CSS transform property. ([#11310](https://github.com/mapbox/mapbox-gl-js/pull/11310))
+* Fix `getBounds` sometimes returning invalid `LngLat` when zooming on a map with terrain. ([#11339](https://github.com/mapbox/mapbox-gl-js/pull/11339))
```
Comment (SnailBones) [inline CHANGELOG.md:18]: Ted's investigation was helpful in solving this bug
```suggestion
* Fix `getBounds` sometimes returning invalid `LngLat` when zooming on a map with terrain. ([#11339](https://github.com/mapbox/mapbox-gl-js/pull/11339)) (h/t [@ted-piotrowski](https://github.com/ted-piotrowski))
```

**Review by SnailBones — CHANGES_REQUESTED** [review]
A few typos

File: CHANGELOG.md (hunk 10 above)
Comment (avpeery) [reply CHANGELOG.md]: I think I fixed this one already! Committed the other changes!

**Review by SnailBones — APPROVED** [review]
:+1:

**Pull request merged**